### PR TITLE
Allow to set active and disabled class also to .page-link

### PR DESCRIPTION
**Pull request by thewebartisan7**
Sometimes we can set `.active` class only to link tag and not parent.
Since active status style is applied only to `.page-link` and not `.page-item`, would also make more sense to just add the active class to `.page-link` itself.

The other way to set `.active` class to `.page-item` still remain, so there is not BC.

Update: allow to set also `.disabled` class to `.page-link`

### CODE REVIEW

File: scss/_pagination.scss [hunk 1]
```diff
@@ -27,6 +27,13 @@
     outline: $pagination-focus-outline;
     box-shadow: $pagination-focus-box-shadow;
   }
+  
+  &.active {
```
Comment (ffoodd) [inline scss/_pagination.scss:31]: This leads to duplicate code. Could you merge both declarations, eg. using `.active, .active > &` as a selector — and then drop the other block in `.page-item`?

**Review by ffoodd — CHANGES_REQUESTED** [review]
Thanks, that makes sense indeed. We just need to avoid code duplication but apart from that, that's good!

File: scss/_pagination.scss (hunk 1 above)
Comment (thewebartisan7) [reply scss/_pagination.scss]: You are right. I didn't think about it. I have update code.

File: scss/_pagination.scss [hunk 2]
```diff
@@ -27,20 +27,22 @@
     outline: $pagination-focus-outline;
     box-shadow: $pagination-focus-box-shadow;
   }
+  
+  &.active,
+  .page-item.active & {
```
Comment (ffoodd) [inline scss/_pagination.scss:32]: `.active > & {` feels more accurate to me, taking direct parenthood into account while decreasing the specificity.

Comment (thewebartisan7) [reply scss/_pagination.scss]: Yes I agree with you. But this specificity it's like current code that we removed:

```scss
.page-item {
  //...
  &.active .page-link {
    z-index: 3;
    color: $pagination-active-color;
    @include gradient-bg($pagination-active-bg);
    border-color: $pagination-active-border-color;
  }
  //...
}
```

Anyway, I have make change as you suggest.

File: scss/_pagination.scss [hunk 3]
```diff
@@ -27,20 +27,21 @@
     outline: $pagination-focus-outline;
     box-shadow: $pagination-focus-box-shadow;
   }
+
```
Comment (ffoodd) [inline scss/_pagination.scss:30]:
```suggestion
```

**Review by ffoodd — CHANGES_REQUESTED** [review]
Drop whitespace to pass linting.

File: scss/_pagination.scss [hunk 4]
```diff
@@ -27,20 +27,20 @@
     outline: $pagination-focus-outline;
     box-shadow: $pagination-focus-box-shadow;
   }
+  &.active,
```
Comment (ffoodd) [inline scss/_pagination.scss:30]:
```suggestion

  &.active,
```

**Review by ffoodd — COMMENTED** [review]
Unwanted dropped line break too.

**Review by ffoodd — APPROVED** [review]
LGTM now 👌 Thanks!

**Comment by ffoodd** [comment]
A simple comment to myself: we should probably move and enhance the disabled state toi (allowing `.disabled` on links, and maybe handling missing `href`/`:link` or something).

**Comment by thewebartisan7** [comment]
> A simple comment to myself: we should probably move and enhance the disabled state toi (allowing `.disabled` on links, and maybe handling missing `href`/`:link` or something).

Yes good point. I think that would be my second issue as I am creating blade components with bootstrap, and I have a link component where I have prop active and disabled. While this are only in page-item and not in page-link. I think that I would come here again for the same reason very soon.  I can add the same thing like for active:

```scss
.page-link {
  //...
  &.disabled,
  .disabled > &  {
    color: $pagination-disabled-color;
    pointer-events: none;
    background-color: $pagination-disabled-bg;
    border-color: $pagination-disabled-border-color;
  }
  //...
}

  ```

**Comment by thewebartisan7** [comment]
Added for the `disabled` class same like for the `active`.

**Review by ffoodd — APPROVED** [review]
Looks good to me, thanks!

We could make states classless using `[aria-current]` for active and `:not([href])` for disabled, but that wouldn't match our components conventions. Only leaving it hère for records.

**Comment by thewebartisan7** [comment]
I think is worth to consider to apply the same thing also for breadcrumb component, where the active class is set to `breadcrumb-item`.

I see that in docs the active item has not even the link tag, and that a`aria-current="page"` is set to breadcrumb-item.

But according to [wai docs](https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html) `aria-current="page"` should be applied to last link. 

Below is a possible fix to archive the same style without breaking current way:

```scss
.breadcrumb-item {
  // ...
  &.active,
  > .active {
    color: $breadcrumb-active-color;
  }

  > .active {
    text-decoration: none;
    cursor: default;
    pointer-events: none;
  }
}
```

And the markup could be:

```html
<nav aria-label="breadcrumb">
  <ol class="breadcrumb">
    <li class="breadcrumb-item">
      <a href="/docs" class="">Home
      </a>
    </li>
    <li class="breadcrumb-item">
      <a href="/docs/components" class="">Components
      </a>
    </li>
    <li class="breadcrumb-item">
      <a href="/docs/components/breadcrumb" aria-current="page" class="active">Breadrumb
      </a>
    </li>
  </ol>
</nav>
```

What do you think? In case you agree, let me know if you want to open a new PR or add in existing.

I checked others components, and I don't see any other that add active class and/or aria-current to a no-link tag.

**Comment by ffoodd** [comment]
Regarding `aria-current`, the [ARIA design pattern](https://www.w3.org/TR/wai-aria-practices/#breadcrumb) says that if current element is not a link, `aria-current` is optional, meaning it doesn't need to be a link to match the pattern expectations.

If a link is used, I don't think it should be disabled through CSS.

**Review by mdo — APPROVED** [review]

**Comment by mdo** [comment]
@thewebartisan7 Can you resolve conflicts when you have time?

**Comment by thewebartisan7** [comment]
@mdo conflict solved.

**Review by ffoodd — APPROVED** [review]

**Pull request merged**